pkg/solidity-utils/contracts/math/BasePoolMath.sol

```diff
@@ -186,20 +195,25 @@ library BasePoolMath {
         uint256 exactBptAmountOut,
         uint256 totalSupply,
         uint256 swapFeePercentage,
-        function(uint256[] memory, uint256, uint256) external view returns (uint256) computeBalance
+        IBasePool pool
     ) internal view returns (uint256 amountInWithFee, uint256[] memory swapFeeAmounts) {
         // Calculate new supply after minting exactBptAmountOut
         uint256 newSupply = exactBptAmountOut + totalSupply;
+
         // Calculate the initial amount of the input token needed for the desired amount of BPT out
         // "divUp" leads to a higher "newBalance", which in turn results in a larger "amountIn".
         // This leads to receiving more tokens for the same amount of BPT minted.
-        uint256 newBalance = computeBalance(currentBalances, tokenInIndex, newSupply.divUp(totalSupply));
+        uint256 invariantRatio = newSupply.divUp(totalSupply);
```

Comment:
I don't understand this line. Why is the division of balances an invariant?

Reply:
that's by definition :) it already was like that, only that it was implicit (check the last parameter name for `computeBalance`).

Invariants must grow linearly with BPT supply, otherwise the system doesn't make sense. Also check proportional math for reference: if you double the balances, you'll double the BPT supply and the invariant.